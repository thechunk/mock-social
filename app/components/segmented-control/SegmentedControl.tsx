import * as React from 'react';
import {Children, FunctionComponent, ReactElement, useState} from 'react';
import {View} from 'react-native';
import SegmentedControlContext from './SegmentedControlContext';

const TabBar = require('./TabBar').default;
const Content = require('./Content').default;


const SegmentedControl: FunctionComponent<ISegmentedControl> = (props) => {
    const [selectedTab, setSelectedTab] = useState<number>(0);

    let error = false;
    Children.forEach<ReactElement<any, any>>(props.children as ReactElement[], (child) => {
        if (!child || (child.type != TabBar && child.type != Content)) {
            error = true;
            return console.error(`No children defined or invalid child type: ${child.type.displayName}`);
        }
    });

    if (error) return null;
    return (
        <SegmentedControlContext.Provider value={{selectedTab, setSelectedTab}}>
            <View>
                {props.children}
            </View>
        </SegmentedControlContext.Provider>
    )
};

export default SegmentedControl;
