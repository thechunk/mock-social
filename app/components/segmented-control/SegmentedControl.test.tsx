import * as React from 'react';
import {shallow} from "enzyme";
import SegmentedControl from "./SegmentedControl";

describe('SegmentedControl', () => {
    it('renders correctly', () => {
        expect(shallow(<SegmentedControl />)).toMatchSnapshot();
    });
});