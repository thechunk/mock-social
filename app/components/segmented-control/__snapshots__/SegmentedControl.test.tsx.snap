// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedControl renders correctly 1`] = `
<ContextProvider
  value={
    Object {
      "selectedTab": 0,
      "setSelectedTab": [Function],
    }
  }
>
  <View />
</ContextProvider>
`;
